src/runtime/crdt/tests/crdt-collection-test.ts

```diff
@@ -157,69 +167,181 @@ describe('CRDTCollection', () => {
+    assert.isTrue(set2.applyOperation(removeOp('removed by b added by a', 'b', {b: 2, c: 5})));
+    assert.isTrue(set2.applyOperation(addOp('removed by a added by b', 'b', {b: 3, c: 5})));
+
+    const {modelChange, otherChange} = set1.merge(set2.getData());
+
+    const expectedVersion = {a: 3, b: 3, c: 5};
+    assert.deepEqual(modelChange, {
+      changeType: ChangeType.Model,
+      modelPostChange: {
+        values: {
+          'kept by both': {value: {id: 'kept by both'}, version: {c: 1}},
+          'removed by a added by b': {value: {id: 'removed by a added by b'}, version: {b: 3, c: 5}},
+          'removed by b added by a': {value: {id: 'removed by b added by a'}, version: {a: 3, c: 5}},
+          'added by a': {value: {id: 'added by a'}, version: {a: 1, c: 5}},
+          'added by b': {value: {id: 'added by b'}, version: {b: 2, c: 5}},
+          'added by both': {value: {id: 'added by both'}, version: {a: 2, b: 1, c: 5}},
+        },
+        version: expectedVersion
+      },
     });
-    set1.applyOperation({
-      type: CollectionOpTypes.Add,
-      added: {id: 'two'},
-      clock: {me: 2},
-      actor: 'me'
+    assert.deepEqual(otherChange, {
+      changeType: ChangeType.Operations,
+      operations: [{
+        type: CollectionOpTypes.FastForward,
+        added: [
+          [{id: 'added by both'}, {a: 2, b: 1, c: 5}],
+          [{id: 'removed by b added by a'}, {a: 3, c: 5}],
+          [{id: 'added by a'}, {a: 1, c: 5}],
+        ],
+        removed: [
+          {id: 'removed by a'},
+        ],
+        oldClock: {b: 3, c: 5},
+        newClock: expectedVersion,
+      }],
     });
-    const set2 = new CRDTCollection<{id: string}>();
-    set2.applyOperation({
-      type: CollectionOpTypes.Add,
-      added: {id: 'three'},
-      clock: {you: 1},
-      actor: 'you'
+  });
+
+  describe('fast-forward operations', () => {
+    it('rejects fast-forward ops which begin in the future', () => {
+      const set = new CRDTCollection<{id: string}>();
+      
+      assert.isFalse(set.applyOperation({
+        type: CollectionOpTypes.FastForward,
+        added: [],
+        removed: [],
+        oldClock: {a: 5},  // > 0
+        newClock: {a: 10},
+      }));
     });
-    set2.applyOperation({
-      type: CollectionOpTypes.Add,
-      added: {id: 'one'},
-      clock: {you: 2},
-      actor: 'you'
+
+    it('rejects fast-forward ops which end in the past', () => {
+      const set = new CRDTCollection<{id: string}>();
+      
+      // Add some initial elements.
+      assert.isTrue(set.applyOperation(addOp('one', 'me', {me: 1})));
+      assert.isTrue(set.applyOperation(addOp('two', 'me', {me: 2})));
+      assert.isTrue(set.applyOperation(addOp('three', 'me', {me: 3})));
+      
+      assert.isFalse(set.applyOperation({
+        type: CollectionOpTypes.FastForward,
+        added: [],
+        removed: [],
+        oldClock: {a: 1},
+        newClock: {a: 2},  // < 3
```

Comment:
I think this is failing because oldClock.a is not zero. Were you intending to test something like:
oldClock: {me:3} newClock: {me:2}